strawberry/django/test/client.py

```diff
@@ -0,0 +1,70 @@
+import json
+from typing import Any, Dict, Optional
+
+from typing_extensions import Literal
+
+from strawberry.test import BaseGraphQLTestClient, Response
+
+
+class GraphQLTestClient(BaseGraphQLTestClient):
+    def query(
+        self,
+        query: str,
+        variables: Optional[Dict[str, Any]] = None,
+        headers: Optional[Dict[str, Any]] = None,
+        asserts_errors: Optional[bool] = True,
+        format: Literal["multipart", "json"] = "json",
+        **kwargs,
+    ) -> Response:
+        body: Any = {"query": query}
+
+        if variables:
+            body["variables"] = variables
+
+        if format == "multipart":
+            assert variables is not None
+
+            # We have fo map the variables keywords with the files provide in `kwargs`
+            ref_variable = list(variables.keys())[0]
+            # In case of folders the variables will look like
+            # `{"folder": {"files": ...]}}`
+            if isinstance(variables.get(ref_variable), dict):
+                ref_variable += f".{list(variables[ref_variable].keys())[0]}"
+
+            # If the variable is an array of files we must number the keys
+            if len(kwargs) == 1:
+                file_map = json.dumps({ref_variable: [f"variables.{ref_variable}"]})
+            else:
+                file_map = json.dumps(
+                    {
+                        k: [f"variables.{ref_variable}.{index}"]
+                        for index, k in enumerate(kwargs)
+                    }
+                )
```

Comment:
I think theres's a bit too much going on on some of these lines, even with the comments. Can you split things into multiple lines with some intermediary variables? e.g.

```python
if isinstance(variables.get(ref_variable), dict):
    var_with_descriptive_name = next(iter(variables[ref_variable]))
    ref_variable += f".{var_with_descriptive_name}"
```

Reply:
I put this logic into a separate method and added comments... it's still a bit complicated but hope now is at least a bit more clear 😒